aiida/manage/tests/__init__.py

```diff
@@ -272,14 +272,11 @@ def profile_dictionary(self):
             'database_name': self.profile_info.get('database_name'),
             'database_username': self.profile_info.get('database_username'),
             'database_password': self.profile_info.get('database_password'),
-            'broker_protocol': self.profile_info.get('broker_protocol'),
-            'broker_username': self.profile_info.get('broker_username'),
-            'broker_password': self.profile_info.get('broker_password'),
-            'broker_host': self.profile_info.get('broker_host'),
-            'broker_port': self.profile_info.get('broker_port'),
-            'broker_virtual_host': self.profile_info.get('broker_virtual_host'),
             'repository_uri': f'file://{self.repo}',
         }
+        for key in ['broker_protocol', 'broker_username', 'broker_password', 'broker_host', 'broker_port']:
```

Comment:
I guess currently these are the only ones regarded as optional, but one could also imagine e.g. the `database_post`  have a default value.

Why not use this loop for all keys of the dictionary?

Reply:
Just to note, 
for the broker defaults, they currently come from `BROKER_DEFAULTS`: https://github.com/aiidateam/aiida-core/blob/75310e0bfbf8de02302de76f3e452b38e4164f33/aiida/manage/external/rmq.py#L41, and then the database defaults come from somewhere else.

But really I think all the defaults should derive from the schema defaults (in https://github.com/aiidateam/aiida-core/blob/develop/aiida/manage/configuration/schema/config-v5.schema.json), as happens for the options, i.e. the schema would be the "source of truth" for all defaults.

I just didn't have time to do this with the configuration refactor

Reply:
> Why not use this loop for all keys of the dictionary?

If `null` is invalid for all keys I agree that makes more sense.

Maybe getting the defaults for both the broker and the other options from the schema can be a separate PR?